public IOException rethrowAsIOException() throws IOException{
    IOException newException = new IOException(getMessage());
    newException.initCause(this);
    throw newException;
}